include/GafferArnoldUI/BarndoorVisualiser.h

```diff
@@ -0,0 +1,65 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2017, Image Engine. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef GAFFERARNOLDUI_BARNDOORVISUALISER_H
+#define GAFFERARNOLDUI_BARNDOORVISUALISER_H
+
+#include "IECore/ObjectVector.h"
+#include "IECoreGL/Renderable.h"
+
+#include "GafferSceneUI/StandardLightFilterVisualiser.h"
+
+namespace GafferArnoldUI
+{
+
+class BarndoorVisualiser : public GafferSceneUI::StandardLightFilterVisualiser
+{
+
+public :
+
+	IE_CORE_DECLAREMEMBERPTR( BarndoorVisualiser )
+
+	BarndoorVisualiser();
+	~BarndoorVisualiser();
+
+	virtual IECoreGL::ConstRenderablePtr visualise( const IECore::InternedString &attributeName, const IECore::ObjectVector *shaderVector, const IECore::Shader *lightShader, IECoreGL::ConstStatePtr &state ) const;
```

Comment:
I guess that raises the question whether I should aim for getting this merged into 34 or just the next release that supports c++11? I had initially planned to eventually create a PR that converts some of the non-cpp11 stuff here into more modern cpp. I think I'm also using NULL instead of nullptr somewhere in the code...